aws/rust-runtime/aws-sig-auth/src/signer.rs

```diff
@@ -156,28 +161,38 @@ impl SigV4Signer {
             settings,
         };
 
-        // A body that is already in memory can be signed directly. A  body that is not in memory
-        // (any sort of streaming body) will be signed via UNSIGNED-PAYLOAD.
-        let signable_body = request_config
-            .payload_override
-            // the payload_override is a cheap clone because it contains either a
-            // reference or a short checksum (we're not cloning the entire body)
-            .cloned()
-            .unwrap_or_else(|| {
-                request
-                    .body()
-                    .bytes()
-                    .map(SignableBody::Bytes)
-                    .unwrap_or(SignableBody::UnsignedPayload)
-            });
-
-        let (signing_headers, signature) =
-            calculate_signing_headers(request, signable_body, &sigv4_config)?.into_parts();
-        for (key, value) in signing_headers {
-            request
-                .headers_mut()
-                .append(HeaderName::from_static(key), value);
+        let (memo, signature) = {
```

Comment:
nit: I have no idea what memo is here

Reply:
Yeah, memo is not the right name for this. I had a hard time naming it, and at a certain point, I just had pick something and move on. It's a struct with instructions on how to mutate a request so that it's signed, in the form of query params and headers to add to it. Maybe `SigningInstructions` paired with a local named `instructions` would be better.